monai/handlers/confusion_matrix.py

```diff
@@ -86,20 +94,34 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
         if len(output) != 2:
             raise ValueError(f"output must have length 2, got {len(output)}.")
         y_pred, y = output
-        score, not_nans = self.confusion_matrix(y_pred, y)
-        not_nans = int(not_nans.item())
-
-        # add all items in current batch
-        self._sum += score.item() * not_nans
-        self._num_examples += not_nans
-
-    @sync_all_reduce("_sum", "_num_examples")
-    def compute(self) -> float:
+        if self.compute_sample is True:
+            score, not_nans = self.confusion_matrix(y_pred, y)
+            not_nans = int(not_nans.item())
+
+            # add all items in current batch
+            self._sum += score.item() * not_nans
+            self._num_examples += not_nans
+        else:
+            confusion_matrix = self.confusion_matrix(y_pred, y)
+            confusion_matrix, _ = do_metric_reduction(confusion_matrix, MetricReduction.SUM)
+            self._total_tp += confusion_matrix[0].item()
+            self._total_fp += confusion_matrix[1].item()
+            self._total_tn += confusion_matrix[2].item()
+            self._total_fn += confusion_matrix[3].item()
+
+    @sync_all_reduce("_sum", "_num_examples", "_total_tp", "_total_fp", "_total_tn", "_total_fn")
```

Comment:
Could you please add a distributed test or at least do it locally to make this `sync_all_reduce` can work fine?
Refer to this test code: https://github.com/Project-MONAI/MONAI/blob/master/tests/test_handler_rocauc_dist.py

Thanks.